tests/tasks.py

```diff
@@ -293,3 +292,40 @@ def mytask(longer_arg):
             eq_(arg.names, ('longer-arg', 'l'))
             eq_(arg.attr_name, 'longer_arg')
             eq_(arg.name, 'longer_arg')
+
+        def decorated_function_names_are_preserved(self):
```

Comment:
Cool, good to know. Thanks!

--- Original Message ---

From: "Jeff Forcier" <notifications@github.com>
Sent: May 22, 2015 5:23 PM
To: "pyinvoke/invoke" <invoke@noreply.github.com>
Cc: "Tye Wang" <tyewang@outlook.com>
Subject: Re: [invoke] Add handling of decorators when parsing arguments (#246)

> @@ -293,3 +292,40 @@ def mytask(longer_arg):
>              eq_(arg.names, ('longer-arg', 'l'))
>              eq_(arg.attr_name, 'longer_arg')
>              eq_(arg.name, 'longer_arg')
> +
> +        def decorated_function_names_are_preserved(self):

That setup lets you load individual test files with `-m` (so e.g. `inv test -m runners` to run `tests/runners.py`). spec lacks support for individual tests, which is annoying and something I'll probably try to get working soon.

---
Reply to this email directly or view it on GitHub:
https://github.com/pyinvoke/invoke/pull/246/files#r30931654